### DISCUSSION TITLE
How can I format a fat32 disk

### DISCUSSION THREAD

**Discussion by MSTendo64** [CosmosOS/Cosmos#3021] [Q&A] [answered]
I want to make an installer for my OS, but I don't understand how you can format a disk directly from the installer

**Comment by 9xbt** [comment]
read the friendly manual

**Comment by retrievaaaa** [comment]
read the docs

**Comment by Gabolate** [comment]
> I want to make an installer for my OS, but I don't understand how you can format a disk directly from the installer

What kind of installer? to boot your OS directly from the harddisk or just to store some settings?

**Comment by MSTendo64** [comment]
@Gabolate an installer that formats the disk and allows you to save configs and just files because I want to run the OS on real hardware

**Comment by Gabolate** [comment, marked as answer]
@MSTendo64 Just use VFS, `fs.Disks[diskIndex].FormatPartition(partitionIndex, "FAT32");`
And pls, remember to read the docs and make tests before making a github issue
